Add constants for magic numbers throughout the code

thesprintingturtle opened the pull request:
This adds constants in the RobotMap class to consolidate things like CAN IDs, PWM Ports, DIO Ports, and Analog Ports, as well as some subsystem-specific constants.

To Test:
1. Build and deploy on robot.
2. Ensure normal operation.

Commit:
Partially Comleted Set of Constants

coderkevin committed:
Changed const suffixes to prefixes

This changes the suffixes such as "_CAN_ID" to prefixes like "CAN_ID_"
which provides for better intellisense lookups.

coderkevin committed:
Add constants for controller ids

coderkevin committed:
Adjust formatting for RobotMap

coderkevin reviewed (approved):
This looks good to me, and I test built it and it built fine.

thesprintingturtle commented:
Not sure how to insert a code block like you usually do, but I would like to put the default return value of the two functions in VisionSub in the Vision category.

Commit:
Added Vision Constants

Added constants to RobotMap that store the default reurn values for getTargetXAngle and getTargetArea in VisionSub.

coderkevin reviewed (approved):
New commit looks fine. Merging.
